### PR TITLE
🧑‍💻 Migrate to Deno for development

### PR DESCRIPTION
**Pull request by cowboyd**
## Motivation

There are two main reasons that we want to migrate to Deno:

1. The TS types that we currently ship with our interactors packages are broken, and rather than maintain and build the npm packages by hand, we can just use DNT to build them for us knowing that they'll be maximally compatible with NPM going forward.
2. We're planning on using Deno to build the interactors CLI, so it would be nice to have that in place first.

## Approach

Migrate to a single Deno workspace, and then use covector for change management and publication. Releases will represented via a tag, and then publication to package registries can happen independently.

Shout out to @wKich for doing the heavy lifting on this!

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -1,62 +1,19 @@
...
-    "test:unit": "tsx --test --test-reporter spec --no-deprecation test/**/*.test.ts",
-    "test:types": "tsd -f ./types/*.test-d.ts",
-    "test": "yarn test:unit && yarn test:types",
-    "docs": "rm -rf docs && yarn typedoc --options typedoc.json",
-    "prepack": "tsc --build ./tsconfig.build.json && yarn prepack:es2015 && yarn prepack:commonjs",
-    "prepack:es2015": "tsc --project ./tsconfig.build.json --outdir dist/esm --module es2015",
-    "prepack:commonjs": "tsc --project ./tsconfig.build.json --outdir dist/cjs --module commonjs"
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/thefrontside/interactors.git"
```
Comment (jbolda) [inline packages/core/package.json:10]: If we are keeping the package.json, do we want to add the directory? I don't know if DNT will handle that or not.
https://docs.npmjs.com/cli/v10/configuring-npm/package-json#repository

File: packages/core/src/element/click.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
-import userEvent from '@testing-library/user-event'
+// @ts-ignore 123
```
Comment (jbolda) [inline packages/core/src/element/click.ts:1]: Should we prefer `@ts-expect-error` to allow for Deno to inform us when it isn't needed?

File: packages/keyboard/package.json [hunk 3]
```diff
@@ -1,52 +1,18 @@
...
-    "typedoc": "^0.25.13",
-    "typescript": "~5.4.5"
+  "bugs": {
+    "url": "https://github.com/thefrontside/interactors/issues"
   },
-  "volta": {
-    "extends": "../../package.json"
+  "engines": {
+    "node": ">= 16"
   }
```
Comment (jbolda) [inline packages/keyboard/package.json:17]: `sideEffects` config here?

File: packages/with-cypress/package.json [hunk 4]
```diff
@@ -22,7 +22,7 @@
     "cypress": ">=6.0.0 <13.0.0"
   },
   "scripts": {
-    "start": "cd ../../sample/ && yarn && yarn start -Y -p 3000",
+    "start": "cd ./cypres/sample/ && yarn && yarn start -Y -p 3000",
```
Comment (jbolda) [inline packages/with-cypress/package.json:25]: spelling?

**Review by jbolda — APPROVED** [review]
Looking lovely! Just a few comments on the `package.json` which we should check in every file / package.

File: packages/core/package.json (hunk 1 above)
Comment (cowboyd) [reply packages/core/package.json]: Great idea... added.

File: packages/keyboard/package.json (hunk 3 above)
Comment (cowboyd) [reply packages/keyboard/package.json]: `sideEffects` is not hard set to true.

File: packages/with-cypress/package.json [hunk 5]
```diff
@@ -8,7 +8,7 @@
   "repository": "https://github.com/thefrontside/interactors.git",
   "homepage": "https://frontside.com/interactors",
   "author": "Frontside Engineering <engineering@frontside.com>",
-  "license": "MIT",
+n  "license": "MIT",
```
Comment (wKich) [inline packages/with-cypress/package.json:11]: Seems it's typo

Comment (cowboyd) [reply packages/with-cypress/package.json:11]: Nooooooooo! 😬

**Pull request merged**